### ISSUE TITLE
Issue hosting IPT in docker container

### ISSUE THREAD

**Issue by AJ1985** [gbif/docker-ipt#4]
Hi, 

I have been trying to host IPT in docker container on windows for last few days with no luck. The container get started but on accessing on localhost:8080 give internal server 500 error. Docker logs on containers gives below

21-Jun-2018 05:50:09.368 INFO [http-nio-8080-exec-1] com.google.inject.struts2.Struts2Factory.createInjector Loading struts2 Guice support...
DEBUG [org.gbif.ipt.config.IPTModule] - Loaded supported jdbc driver information from jdbc.properties
INFO  [org.gbif.ipt.config.ConfigManagerImpl] - IPT DataDir configured - loading its configuration
INFO  [org.gbif.ipt.config.ConfigManagerImpl] - Reading DATA DIRECTORY: /usr/local/ipt
INFO  [org.gbif.ipt.config.ConfigManagerImpl] - Loading IPT config ...
DEBUG [org.gbif.ipt.config.AppConfig] - Loaded default configuration from application.properties in classpath
WARN  [org.gbif.ipt.config.AppConfig] - DataDir configured, but user configuration doesnt exist: /usr/local/ipt/config/ipt.properties
WARN  [org.gbif.ipt.config.AppConfig] - Registry lock file not found meaning the DataDir is NOT locked to a registry yet!
INFO  [org.gbif.ipt.config.ConfigManagerImpl] - Reloading log4j settings ...
INFO  [org.gbif.ipt.config.ConfigManagerImpl] - Setting logging dir to /usr/local/ipt/logs

I believe the error is because of missing user configuration. Should not system still load if configuration is missing?

Can you please suggest what can be done to fix this and load system even if user configurations doesn't exists?

Regards,
Ajay

**Comment by mdoering** [comment]
Have you tried to access the container by its own IP as explained here?
https://github.com/gbif/docker-ipt#run-docker-image-locally

Those two little steps work fine for me. The IPT setup screen just uses a bad Base URL so the CSS is not loaded right. But you can fix that easily by adjusting the detected base URL to the right one (for me it was just stripping off the session id parameter)

**Comment by mdoering** [comment]
btw, I have the same logs and they are perfectly fine, not indicating any error

**Comment by AJ1985** [comment]
Hi Mdoering, 

Thank you for the quick reply and I am sorry for taking this long to reply. 

I was deploying it corporate network. I found that during deployment process application tries to access internet to download vocabulary.  Since application didn't have the proxy details the request to download vocabulary would fails hence opening application would throw null pointer exception. 

To solve this, I copied ipt.properties file from the older deployment with proxy details into /usr/local/ipt/config folder and restarted the container.

Thanks again for the quick response.
